specs/fault-proof/stage-one/anchor-state-registry.md

```diff
@@ -0,0 +1,214 @@
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Definitions](#definitions)
+  - [Dispute Game](#dispute-game)
+  - [Registered Game](#registered-game)
+  - [Respected Game](#respected-game)
+  - [Blacklisted Game](#blacklisted-game)
+  - [Retired Game](#retired-game)
+  - [Proper Game](#proper-game)
+- [Assumptions](#assumptions)
+  - [aASR-001: Dispute Game contracts properly report important properties](#aasr-001-dispute-game-contracts-properly-report-important-properties)
+    - [Mitigations](#mitigations)
+  - [aASR-002: Dispute Game Factory properly reports its created games](#aasr-002-dispute-game-factory-properly-reports-its-created-games)
+    - [Mitigations](#mitigations-1)
+  - [aASR-003: OptimismPortal properly reports respected game type, blacklist, and retirement time](#aasr-003-optimismportal-properly-reports-respected-game-type-blacklist-and-retirement-time)
+    - [Mitigations](#mitigations-2)
+- [Invariants](#invariants)
+  - [iASR-001: Games are represented as Proper Games accurately](#iasr-001-games-are-represented-as-proper-games-accurately)
+    - [Impact](#impact)
+    - [Dependencies](#dependencies)
+- [Function Specification](#function-specification)
+  - [isGameRegistered](#isgameregistered)
+  - [isGameRespected](#isgamerespected)
+  - [isGameBlacklisted](#isgameblacklisted)
+  - [isGameRetired](#isgameretired)
+  - [isGameProper](#isgameproper)
+  - [getAnchorRoot](#getanchorroot)
+  - [anchors](#anchors)
+  - [tryUpdateAnchorState](#tryupdateanchorstate)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `AnchorStateRegistry` was designed as a registry where `DisputeGame` contracts could store and
+register their results so that these results could be used as the starting states for new
+`DisputeGame` instances. These starting states, called "anchor states", allow new `DisputeGame`
+contracts to use a newer starting state to bound the size of the execution trace for any given
+game.
+
+We are generally aiming to shift the `AnchorStateRegistry` to act as a unified source of truth for
+the validity of `DisputeGame` contracts and their corresponding root claims. This specification
+corresponds to the first iteration of the `AnchorStateRegistry` that will move us in this
+direction.
+
+## Definitions
+
+### Dispute Game
+
+> See [Fault Dispute Game](fault-dispute-game.md)
+
+A Dispute Game is a smart contract that makes a determination about the validity of some claim. In
+the context of the OP Stack, the claim is generally assumed to be a claim about the value of an
+output root at a given L2 block height.
+
+### Registered Game
+
+A Dispute Game is considered to be a **Registered Game** if the game contract was created by the
+system's `DisputeGameFactory` contract.
+
+### Respected Game
+
+A Dispute Game is considered to be a **Respected Game** if the game contract's game type is
```

Comment:
This is a self referential definition. A respected game is a respected game if its a respected game. How does the respected game **functionally** impact the system?

Reply:
I don't want to bloat the scope of this PR and have to write a spec for the OptimismPortal. The OptimismPortal currently holds the definition for what a "respected" game type is and the AnchorStateRegistry is piggybacking on this definition.

Respected game type functionally impacts the system because it is used as part of the definition for Proper Game, which is used as part of `tryUpdateAnchorState`

Reply:
Basically what I'm saying here is that OptimismPortal independently has a concept of a "respected game type" and the AnchorStateRegistry defines the concept of a "Respected Game" as a game that has a game type that matches the "respected game type" defined in the OptimismPortal. So this definition is not circular. It just happens to share the same wording.

Reply:
Without this context, its impossible to know its not circular. I recommend using a hyperlink to the place where it is actually defined

Reply:
I honestly don't think this is defined anywhere in the specs

Reply:
I added some extra context in a separate section called `Respected Game Type`, lmk what you think